### ISSUE TITLE
Functionality to handle variable C in CCPoissonPointRelaxationFACOperator 

### ISSUE THREAD

**Issue by Raamkrishnan** [IBAMR/IBAMR#1135]
Hi,
I noticed that ```CCPoissonPointRelaxationFACOperator ``` class does not have any functionality to handle the variable C coefficient. However, I want to have that functionality for the turbulence code to simulate test problems with multiple levels. I can add that functionality. Before doing that, is there any design that I need to follow?

**Comment by boyceg** [comment]
It would be great to support this use case.

I am not at my workstation right, so I cannot confirm, but I think you should be able to do this without any change in the public or protected interfaces — specifically, I think the Poisson specifications object that is used to pass in coefficients should allow specifying either spatially constant or spatially varying coefficients. Do you foresee any changes that would be needed to the interface for your use case?

**Comment by amneetb** [comment]
I think it won't require any change to the API of the class. Some extra code needs to be added like Fortran smoothers that take variable 'C' patch data index, and some `TBOX_ERROR`s need to be removed which check for constant-ness of `C`. All of this needs to be done within the class itself (I think).

**Comment by boyceg** [comment]
Sounds good --- once you have some initial code, I would suggest to put in a PR (feel free to tag with "WIP" and/or "Do not merge") to build up the functionality.

**Comment by drwells** [comment]
Of course, if pieces are ready to merge (like this), then life is easier for everyone if we put them in once they are done.
